pymongo/topology_description.py

```diff
@@ -263,21 +263,24 @@ def apply_selector(self, selector, address=None, custom_selector=None):
                                              selector.min_wire_version,
                                              common_wv))
 
+        if isinstance(selector, _AggWritePref):
+            selector.selection_hook(self)
```

Comment:
Initially I was expecting to place the "selection_hook" logic inside `_AggWritePref.__call__` but it turns out that we skip calling the selector in many cases. In fact the only time we actually do call the selector is for replica sets. I attempted to refactor this code to actually call the selector in all cases but that doesn't work either because read preference selectors are written to only work with replica sets.

So "selection_hook" is what I came up with instead.